build.gradle

```diff
@@ -90,6 +90,36 @@ application {
   mainClass = 'net.sourceforge.kolmafia.KoLmafia'
 }
 
+task cleanDist(type: Delete) {
+  onlyIf {
+    file('dist').exists()
+  }
+  inputs.dir('dist')
+  outputs.upToDateWhen { true }
+  doLast {
+    file('dist').eachFile(groovy.io.FileType.FILES) { File file ->
+      if (file.name.startsWith('KoLmafia-') && file.name.endsWith('.jar')) {
+          delete file
+      }
+    }
+  }
+}
+
+task pruneDist(type: Delete) {
+  onlyIf {
+    file('dist').exists()
+  }
+  inputs.dir('dist')
+  outputs.upToDateWhen { true }
+  doLast {
+    file('dist').eachFile(groovy.io.FileType.FILES) { File file ->
+      if (file.name.startsWith('KoLmafia-') && file.name.endsWith('.jar') && !file.name.contains(project.getVersion())) {
```

Comment:
IMO we should prune KoLmafia-12345-M.jar too, which suggests we should check that !file.name.equals("KoLmafia-" + project.getVersion() + ".jar"), or !endsWith(getVersion() + ".jar").

(In addition to the startsWith / endsWith checks you have, to make sure we only nuke jars.)

Other than that, this change looks good.

Reply:
done